families/sawtooth_validator_registry/validator_registry/src/validator_registry_tp_verifier.rs

```diff
@@ -0,0 +1,212 @@
+        error!("Verification report signature does not match");
+        return Err(ApplyError::InternalError(
+            "Verification report signature does not match".to_string(),
+        ));
+    }
+
+    // Convert verification_report json into HashMap
+    let verification_report_tmp_value: serde_json::Value =
+        serde_json::from_str(verification_report)
+            .expect("Error reading verification report as Json");
+    let verification_report_dict = verification_report_tmp_value
+        .as_object()
+        .expect("Error reading verification report as Key Value pair");
+    // Verify that the verification report meets the following criteria:
+    // Includes an ID field.
+    if !verification_report_dict.contains_key("id") {
+        error!("Verification report does not contain id field");
+        return Err(ApplyError::InternalError(
+            "Verification report does not contain id field".to_string(),
+        ));
+    }
+    // Includes an EPID psuedonym.
+    if !verification_report_dict.contains_key("epidPseudonym") {
+        error!("Verification report does not contain an EPID psuedonym");
+        return Err(ApplyError::InternalError(
+            "Verification report does not contain an EPID psuedonym".to_string(),
+        ));
+    }
+    // Verify that the verification report EPID pseudonym matches the anti-sybil ID
+    let epid_pseudonym = verification_report_dict
+        .get("epidPseudonym")
+        .expect("Error reading epidPseudonym from verification report")
+        .as_str()
+        .expect("Error converting epidPseudonym as string reference");
+    if epid_pseudonym != signup_info.anti_sybil_id {
+        error!(
+            "The anti-sybil ID in the verification report {} does not match the one \
+             contained in the signup information {}",
+            epid_pseudonym, signup_info.anti_sybil_id
+        );
+        return Err(ApplyError::InternalError(
+            "The anti-sybil ID in the verification report does not match in \
+             the signup information"
+                .to_string(),
+        ));
+    }
+    // Includes an enclave quote.
+    if !verification_report_dict.contains_key("isvEnclaveQuoteBody") {
+        error!("Verification report does not contain enclave quote body");
+        return Err(ApplyError::InternalError(
+            "Verification report does not contain enclave quote body".to_string(),
+        ));
+    }
+    // The ISV enclave quote body is base 64 encoded
+    let _enclave_quote = verification_report_dict
+        .get("isvEnclaveQuoteBody")
+        .expect("Error reading isvEnclaveQuoteBody from verification report");
+    // The report body should be SHA256(SHA256(OPK)|PPK)
+    let hash_input = format!(
+        "{}{}",
+        originator_public_key_hash.to_uppercase(),
+        signup_info.poet_public_key.to_uppercase()
+    );
+    let _hash_value = sha256(hash_input.as_bytes());
+    // TODO: Quote verification
```

Comment:
This PR should not include TODO. Please implement the missing piece and remove the TODO comment